### PR TITLE
Add ht insert tests/implementation

### PR DESCRIPTION
**Pull request by TheNeikos**
This is meant to solve #62.

I am not sure if the test I added is sufficient.

### CODE REVIEW

File: tests/ht/ht_test.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include <check.h>
+
+#include <stdlib.h>
+#include "ht.h"
+
+START_TEST (test_ht_insert_data) {
+    struct ht* ht = malloc(sizeof(*ht));
+    ht_init(ht, 4);
+    int data = 5;
+    ck_assert(ht_insert(ht, &data, sizeof(data)));
```
Comment (matthiasbeyer) [inline tests/ht/ht_test.c:10]: I'd like you to compare against something here, please add a compare to `1` or what the expected return value is. Use `ck_assert_int_eq` if this is appropriate!

Comment (TheNeikos) [inline tests/ht/ht_test.c:10]: Doing that now.

**Comment by matthiasbeyer** [comment]
Besides the one point ths looks good so far.

For a TDD workflow, you should add patches to pass the test now (on this very same PR). I'd like to suggest more tests, too: Adding multiple values, adding multiple values which collide in their hashes, etc.

Edit: The title should be adapted, then.

File: tests/ht/ht_test.c (hunk 1 above)
Comment (matthiasbeyer) [inline tests/ht/ht_test.c:10]: Nah, the interface of the ht lacks here.

You need to pass the hash of the element, too. The hashtable should not create the hash of the object (@neithernut right?).

Comment (neithernut) [inline tests/ht/ht_test.c:10]: It should, using a user provided function, see issue #103.

Comment (matthiasbeyer) [inline tests/ht/ht_test.c:10]: So we need interface changes to be able to add a hash function.

**Comment by TheNeikos** [comment]
I will wait for #103 and start from there again. Seems to be the most sensible to me.

**Comment by neithernut** [comment]
You can probably adapt quickly once the changes are done, so keep the branch.

**Comment by matthiasbeyer** [comment]
@TheNeikos #104 changes the interface of `ht_insert()`.

**Comment by matthiasbeyer** [comment]
Please resubmit onto master as soon as the appropriate interface changes are done. My attempt to switch to kind-of multi-branch-workflow was bullshit, we will submit PRs onto master until v0.0.1, please apologize the confusion!

**Pull request closed without merging**